src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -62,7 +62,9 @@ PG_FUNCTION_INFO_V1(replicate_reference_tables);
 Datum
 replicate_reference_tables(PG_FUNCTION_ARGS)
 {
-	EnsureReferenceTablesExistOnAllNodes();
+	Oid shardReplicationModeOid = PG_GETARG_OID(0);
+	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
+	EnsureReferenceTablesExistOnAllNodesExtended(shardReplicationMode);
```

Comment:
This seems like a very important change.

 Do we have any existing tests that shows non-blocking `EnsureReferenceTablesExistOnAllNodesExtended`? Would be good to reference in the PR description.

Reply:
rebalance_table_shards already calls the non-blocking variant by default, but it probably does make sense to cover the UDF call in tests

Reply:
> rebalance_table_shards already calls the non-blocking variant by default, 

Ah, cool, I missed that

Reply:
We had some tests for `replicate_reference_tables()` already, I modified those slightly to test all the shard transfer modes.